rasa/nlu/regex_message_handler.py

```diff
@@ -0,0 +1,292 @@
+                f"Make sure the intent is followed by a proper json object (dict). "
+                f"Continuing without entities. "
+                f"Error: {e}",
+                docs=DOCS_URL_STORIES,
+            )
+            parsed_entities = dict()
+
+        # validate the given entity types
+        entity_types = set(parsed_entities.keys())
+        unknown_entity_types = entity_types.difference(domain.entities)
+        if unknown_entity_types:
+            rasa.shared.utils.io.raise_warning(
+                f"Failed to parse arguments in line '{match.string}'. "
+                f"Expected entities from {domain.entities} "
+                f"but found {unknown_entity_types}. "
+                f"Continuing without unknown entity types. ",
+                docs=DOCS_URL_STORIES,
+            )
+            parsed_entities = {
+                key: value
+                for key, value in parsed_entities
+                if key not in unknown_entity_types
+            }
+
+        # convert them into the list of dictionaries that we expect
+        entities: List[Dict[Text, Any]] = []
+        for entity_type, entity_values in parsed_entities.items():
+            if not isinstance(entity_values, list):
+                entity_values = [entity_values]
+            for entity_value in entity_values:
+                entities.append(
+                    {
+                        ENTITY_ATTRIBUTE_TYPE: entity_type,
+                        # TODO: in the original implementation these are start and
+                        # end indices of the *substring that matches the entity regex*
+                        # ... but it should be indices of tokens to which these
+                        # entities belong. There are no such tokens. Hence, I think we
+                        # should just skip those indices. But can we, or will something
+                        # complain? If something expects indices (despite not being
+                        # able to use them because there's no text), then we could
+                        # instead use more explicit "unknown" indicators (e.g. -1)
+                        # ENTITY_ATTRIBUTE_START: sidx,
+                        # ENTITY_ATTRIBUTE_END: eidx,  # can't be more specific
+                        ENTITY_ATTRIBUTE_VALUE: entity_value,
+                    }
+                )
+        return entities
+
+    @staticmethod
+    def _parse_optional_confidences(match: Match) -> float:
+        """Extracts the optional confidence information from the given pattern match.
+
+        If no confidence is specified, then this method returns the maximum
+        confidence `1.0`.
+        If a confidence is specified but extraction fails, then this method defaults
+        to a confidence of `0.0`.
+
+        Args:
+          match: a match produced by `self.pattern`
+          domain: the domain
+        Returns:
+          some confidence value
```

Comment:
The docstring format seems to not be correct (see other docstrings).

Reply:
looked ok for me - but I've replaced the whitespaces with tabs anyway - better now?